reactor-core/src/main/java/reactor/core/scheduler/SchedulerMetricDecorator.java

```diff
@@ -0,0 +1,58 @@
+package reactor.core.scheduler;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.WeakHashMap;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.BiFunction;
+
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.core.instrument.binder.jvm.ExecutorServiceMetrics;
+
+import reactor.core.Scannable;
+import reactor.core.Scannable.Attr;
+
+import static io.micrometer.core.instrument.Metrics.globalRegistry;
+
+final class SchedulerMetricDecorator
+			implements BiFunction<Scheduler, ScheduledExecutorService, ScheduledExecutorService> {
+
+
+	final WeakHashMap<Scheduler, String>        SEEN_SCHEDULERS          = new WeakHashMap<>();
+	final Map<String, AtomicInteger>            SCHEDULER_DIFFERENTIATOR = new HashMap<>();
+	final WeakHashMap<Scheduler, AtomicInteger> EXECUTOR_DIFFERENTIATOR  = new WeakHashMap<>();
+
+	@Override
+	public synchronized ScheduledExecutorService apply(Scheduler scheduler, ScheduledExecutorService service) {
+		//this is equivalent to `toString`, a detailed name like `parallel("foo", 3)`
+		String schedulerDesc = Scannable
```

Comment:
Why `Desc` if we scan `NAME` attribute? 
"Description" sounds more free-form compared to "name" (I'm thinking about something like "Beautiful red flower" as a description and "Gerbera" as a name)